### PR TITLE
Add instructions for configuring CheckBox as a radio button

### PR DESCRIPTION
**Pull request by Rindbee**

### CODE REVIEW

**Comment by timothyqiu** [comment]
> If the button's derived type is [CheckBox], it will be...

Instead of "hardcoding" `CheckBox` specific behavior in `BaseButton`, I think [ButtonGroup](https://github.com/godotengine/godot/blob/master/doc/classes/ButtonGroup.xml) might be a better place to describe it.

File: doc/classes/ButtonGroup.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	<description>
 		Group of [BaseButton]. The members of this group are treated like radio buttons in the sense that only one button can be pressed at the same time.
 		Every member of the ButtonGroup should have [member BaseButton.toggle_mode] set to [code]true[/code].
+		[b]Note:[/b] A [CheckBox] will be configured as a radio button if a [ButtonGroup] is assigned to it.
```
Comment (YuriSizov) [inline doc/classes/ButtonGroup.xml:9]: This just repeats the first line of this description.

File: doc/classes/CheckBox.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
 	<description>
 		A checkbox allows the user to make a binary choice (choosing only one of two possible options). It's similar to [CheckButton] in functionality, but it has a different appearance. To follow established UX patterns, it's recommended to use CheckBox when toggling it has [b]no[/b] immediate effect on something. For example, it could be used when toggling it will only do something once a confirmation button is pressed.
 		See also [BaseButton] which contains common properties and methods associated with this node.
+		[b]Note:[/b] The [CheckBox] will be configured as a radio button if a [ButtonGroup] is assigned to it. See [member BaseButton.button_group] and [ButtonGroup].
```
Comment (YuriSizov) [inline doc/classes/CheckBox.xml:9]: This is true for any button. I mean, we can repeat it for CheckBox specifically, I guess, but this seems to be missing from the docs on Button/BaseButton too.

Comment (Rindbee) [reply doc/classes/CheckBox.xml:9]: It seems that only the `CheckBox` will have a different appearance when it is configured as a radio button.

File: doc/classes/CheckBox.xml [hunk 3]
```diff
@@ -6,6 +6,7 @@
 	<description>
 		A checkbox allows the user to make a binary choice (choosing only one of two possible options). It's similar to [CheckButton] in functionality, but it has a different appearance. To follow established UX patterns, it's recommended to use CheckBox when toggling it has [b]no[/b] immediate effect on something. For example, it could be used when toggling it will only do something once a confirmation button is pressed.
 		See also [BaseButton] which contains common properties and methods associated with this node.
+		[b]Note:[/b] The [CheckBox] will have a special appearance when it's configured as a radio button. See [member BaseButton.button_group] and [ButtonGroup].
```
Comment (YuriSizov) [inline doc/classes/CheckBox.xml:9]:
```suggestion
		[b]Note:[/b] CheckBox changes its appearance when it's configured as a radio button. See various [code]radio_*[/code] theme properties. To configure CheckBox to act as a radio button, use [member BaseButton.button_group] and [ButtonGroup].
```

File: doc/classes/ButtonGroup.xml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 		Group of Buttons.
 	</brief_description>
 	<description>
-		Group of [BaseButton]. The members of this group are treated like radio buttons in the sense that only one button can be pressed at the same time.
+		Group of [BaseButton]. The members of this group are treated like radio buttons in the sense that only one button can be pressed at the same time. And some members (such as [CheckBox]es) may have a special appearance for this state.
```
Comment (YuriSizov) [inline doc/classes/ButtonGroup.xml:7]:
```suggestion
		Group of [BaseButton]. The members of this group are treated like radio buttons in the sense that only one button can be pressed at the same time. Some types of buttons (such as [CheckBox]) may have a special appearance for this state.
```

**Review by YuriSizov — APPROVED** [review]
A couple of suggestions, but looks good otherwise.

File: doc/classes/BaseButton.xml [hunk 5]
```diff
@@ -49,6 +49,7 @@
 		</member>
 		<member name="button_group" type="ButtonGroup" setter="set_button_group" getter="get_button_group">
 			The [ButtonGroup] associated with the button. Not to be confused with node groups.
+			[b]Note:[/b] The button may be configured as a radio button if a [ButtonGroup] is assigned to it. See [ButtonGroup] for how to configure it.
```
Comment (YuriSizov) [inline doc/classes/BaseButton.xml:52]:
```suggestion
			[b]Note:[/b] The button will be configured as a radio button if a [ButtonGroup] is assigned to it. See [ButtonGroup].
```

This was fine before. The button definitely will act as a radio button, that's the whole point here and what the description of `ButtonGroup` tells us.

Comment (Rindbee) [reply doc/classes/BaseButton.xml:52]: I just feel that it has no effect when `BaseButton.toggle_mode` is not enabled.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.4.

**Pull request merged**